tests/test_tokenization_pegasus.py

```diff
@@ -26,42 +26,54 @@ def setUp(self):
         tokenizer.save_pretrained(self.tmpdirname)
 
     @cached_property
-    def pegasus_large_tokenizer(self):
+    def _large_tokenizer(self):
         return PegasusTokenizer.from_pretrained("google/pegasus-large")
 
-    @unittest.skip("add_tokens does not work yet")
-    def test_swap_special_token(self):
-        pass
-
     def get_tokenizer(self, **kwargs) -> PegasusTokenizer:
-        return PegasusTokenizer.from_pretrained(self.tmpdirname, **kwargs)
+        tok = PegasusTokenizer.from_pretrained(self.tmpdirname, **kwargs)
+        return tok
 
     def get_input_output_texts(self, tokenizer):
         return ("This is a test", "This is a test")
 
-    def test_pegasus_large_tokenizer_settings(self):
-        tokenizer = self.pegasus_large_tokenizer
+    def test_mask_tokens_rust_pegasus(self):
+        rust_tokenizer = self.rust_tokenizer_class.from_pretrained(self.tmpdirname)
+        py_tokenizer = self.tokenizer_class.from_pretrained(self.tmpdirname)
+        raw_input_str = "Let's see which <unk> is the better <unk_token_11> one <mask_1> It seems like this <mask_2> was important </s> <pad> <pad> <pad>"
+        rust_ids = rust_tokenizer([raw_input_str], return_tensors=None, add_special_tokens=False).input_ids[0]
+        py_ids = py_tokenizer([raw_input_str], return_tensors=None, add_special_tokens=False).input_ids[0]
+        # TODO: (Thom, Patrick) - this fails because the rust tokenizer does not know about the <mask_1>, <mask_2>, and those <unk_token_x> yet
```

Comment:
@thomwolf @n1t0 - this test should not fail. The rust tokenizer incorrectly maps the mask tokens and <unk_token_11> at the moment.

Reply:
It's now the only test that fails.

Reply:
Should be solved